Functions/BIDS/xASL_bids_MergeNifti_Merge.m

```diff
@@ -75,17 +83,30 @@
                 end
             end
         end
+        
+	end
 
-        % Check for the path to JSON if existing and keep only the first existing JSON
-        if isempty(firstJSON)
-            [Fpath, Ffile] = xASL_fileparts(NiftiPaths{indexSortedFile(iFile)});
-            pathJSON = fullfile(Fpath,[Ffile '.json']);
-            if exist(pathJSON, 'file')
-                firstJSON = pathJSON;
-            end
-        end
-    end
-
+	%% Merge all the JSONs according to their priority
+	
+	% Create an order in which the JSONs are going to be read
+	[~, priorityListIndex] = sort(priorityList);
+	outputJSON = struct();
+	for iJSON = 1:length(NiftiPaths)
+		% Go through the JSONs in increasing priority
+		[Fpath, Ffile] = xASL_fileparts(NiftiPaths{priorityListIndex(iJSON)});
+		if xASL_exist(fullfile(Fpath,[Ffile '.json']), 'file')
```

Comment:
What happens if this file is in the NIfTIPaths list but not existing?

Reply:
I don't understand?! NiftiPaths contains only NII that we converted and are existing. So all JSONs should be there as well - a missing JSON is very unlikely. But for this unlikely event, I'm checking its existence and if it isn't there it just skips it.

= It shouldn't ever happen, but if it does, it skips it and program doesn't crash.

Reply:
Yes but if it doesn't exist, we DO want the program to crash (or at least we want a warning). If you add a warning in this case, I'm fine and you can resolve.

Reply:
We don't really want crash. But as I say - won't happen ;) Warning added about a missing file.